routers/web/repo/pull.go

```diff
@@ -319,8 +319,37 @@ func PrepareMergedViewPullInfo(ctx *context.Context, issue *models.Issue) *git.C
 	setMergeTarget(ctx, pull)
 	ctx.Data["HasMerged"] = true
 
+	var baseCommit string
+	// Some migrated PR won't have any Base SHA and lose history, try to get one
+	if pull.MergeBase == "" {
+		// If there is a patch file, and it is readable, grab info
+		commitSHA, err := ctx.Repo.GitRepo.ReadPatchCommit(pull.Index)
+		if err != nil {
+			log.Trace("Pull request without history, %v", err)
+		} else {
+			// Get immediate parent of the first commit in the patch, grab history back
+			parentCommit, err := git.NewCommand("rev-list", "-1", "--skip=1", commitSHA).RunInDir(ctx.Repo.GitRepo.Path)
+			// Special case on Git < 2.25 that doesn't fail on immediate empty history
+			if err != nil || parentCommit == "" {
+				log.Info("No known parent commit for PR %d, error: %v", pull.Index, err)
+				// bring at least partial history if it can work
+				parentCommit = commitSHA
+			} else {
+				// Recreate pull head in files
+				err := ctx.Repo.GitRepo.WritePullHead(pull.Index, commitSHA)
+				if err != nil {
+					log.Error("Could not write head file", err)
+				}
+			}
+			baseCommit = strings.TrimSpace(parentCommit)
```

Comment:
Why do we need `TrimSpace` here? In my mind `parentCommit` should be a correct commit ID.

Reply:
I don't know why, but git rev-list returns the commit with a leading whitespace, I checked with different versions and there are always a leading space...

Reply:
Oh I see the problem. Then I think we could make the `Trim` occur right after the NewCommand. I will propose a change suggestion, it's up to you ~